grails-app/services/org/broadinstitute/orsp/PermissionService.groovy

```diff
@@ -10,31 +10,61 @@ class PermissionService implements UserInfo {
+
+        def collaboratorsValues = extraProperties.findAll ({ it.key == IssueExtraProperty.COLLABORATOR }).values()
+        if (collaboratorsValues.size() != 0) {
+            results = collaboratorsValues.first()
+        }
+
+        collaboratorsValues = extraProperties.findAll ({ it.key == IssueExtraProperty.COLLABORATORS }).values()
+        if (results.size() == 0 && collaboratorsValues.size() != 0) {
+            results = collaboratorsValues.first()
+        }
+
+        results
     }
 
     // get issue's pms as a List<String>
-    List<String> getIssuePMs(Issue issue) {
-      issue.extraProperties.findAll ({ it.name == IssueExtraProperty.PM }).collect { property -> property.value }
+    List<String> getIssuePMs(Map<String, List<String>> extraProperties) {
+        List<String> results = new ArrayList<String>()
+        def collaboratorsValues = extraProperties.findAll ({ it.key == IssueExtraProperty.PM }).values()
+        if (collaboratorsValues.size() != 0) {
+            results = collaboratorsValues.first()
+        }
+        results
     }
 
     // get issue's pis as a List<String>
-    List<String> getIssuePIs(Issue issue) {
-      issue.extraProperties.findAll ({ it.name == IssueExtraProperty.PI }).collect { property -> property.value }
+    List<String> getIssuePIs(Map<String, List<String>> extraProperties) {
+        List<String> results = new ArrayList<String>()
+        def collaboratorsValues = extraProperties.findAll ({ it.key == IssueExtraProperty.PI }).values()
+        if (collaboratorsValues.size() != 0) {
+            results = collaboratorsValues.first()
+        }
+        results
     }
 
     // verifies if logged user belongs to some user list ....
     def issueIsForbidden(Issue issue, String userName, boolean isAdmin, boolean isViewer) {
+        Map<String, List<String>> extraProperties = issue.extraPropertiesMap
+        userHasIssueAccess(issue.reporter, extraProperties, userName, isAdmin, isViewer)
+    }
 
-        boolean userHasAccess = (issue.reporter == userName
-                || getIssueCollaborators(issue).indexOf(userName) >= 0
-                || getIssuePMs(issue).indexOf(userName) >= 0
-                || getIssuePIs(issue).indexOf(userName) >= 0
+    def issueIsForbidden(IssueSearchItemDTO issue, String userName, boolean isAdmin, boolean isViewer) {
+        Map<String, List<String>> extraProperties = issue.extraProperties
+        userHasIssueAccess(issue.reporter, extraProperties, userName, isAdmin, isViewer)
+    }
+
+    def userHasIssueAccess(String reporter, Map<String, List<String>> extraProperties, String userName, boolean isAdmin, boolean isViewer) {
+        boolean userHasAccess = (reporter == userName
+                || getIssueCollaborators(extraProperties).indexOf(userName) >= 0
+                || getIssuePMs(extraProperties).indexOf(userName) >= 0
+                || getIssuePIs(extraProperties).indexOf(userName) >= 0
```

Comment:
These boolean checks should be looking at *all* of the PMs, PIs, Collaborators, etc. IndexOf is probably the wrong approach here. I think you want to check `contains` on the list.